server/index.js

```diff
@@ -14,104 +15,68 @@ const loggingMiddleware = require('./logging');
 
 const handle = app.getRequestHandler();
 
-const ssrCache = new Cache({
-  max: 3500,
-  maxAge: 1000 * 60 * 60, // 1hour
-  updateAgeOnGet: true,
-});
-
-const getPageProps = payload => {
-  const string = payload
-    .split('id="__NEXT_DATA__" type="application/json">')[1]
-    .split('</script>')[0];
-  const json = JSON.parse(string);
-  if (json && json.props && json.props.pageProps) {
-    return json.props.pageProps;
-  }
-};
-
-const getCanCache = (pagePath, payload) => {
-  // cache block pages
-  if (pagePath.includes('block') && !pagePath.includes('blocks')) {
-    return true;
-  }
-  const props = getPageProps(payload);
-  if (
-    props &&
-    props.initialData &&
-    props.initialData.transaction &&
-    props.initialData.transaction.tx_status
-  ) {
-    if (props.initialData.transaction.tx_status !== 'pending') {
-      return true;
-    }
-  } else {
-    return false;
-  }
-};
-
-const renderAndCache = app =>
-  async function (req, res, pagePath, queryParams) {
-    const { host } = req.headers;
-    // Define the cache key as you wish here:
-    const key = host + req.url;
-
-    // if page is in cache, server from cache
-    if (ssrCache.has(key)) {
-      console.info('SSR Response from cache for ', key);
-      res.setHeader('x-cache', 'HIT');
-      res.end(ssrCache.get(key));
-      return;
-    }
-
-    try {
-      /**
-       * Override res.end method before sending it to app.renderToHTML
-       * to be able to get the payload (renderedHTML) and save it to cache.
-       */
-      const _resEnd = res.end.bind(res);
-      res.end = function (payload) {
-        // Add here custom logic for when you do not want to cache the page, for example when
-        // the status is not 200
-        const canCache = getCanCache(pagePath, payload);
-
-        if (res.statusCode !== 200) {
-          console.warn('Oops, something is wrong, will skip the cache');
-        } else if (canCache) {
-          ssrCache.set(key, payload);
-        }
-        return _resEnd(payload);
-      };
-      // if not in cache, render the page into HTML
-      res.setHeader('x-cache', 'MISS');
-      console.info('SSR rendering without cache and try caching for ', key);
-      await app.renderToHTML(req, res, pagePath, queryParams);
-    } catch (err) {
-      app.renderError(err, req, res, pagePath, queryParams);
-    }
-  };
-
 app.prepare().then(() => {
   const server = express();
 
-  server.use(createPrometheusMiddleware({ app: server }));
-
   server.get(
     '/_next/',
     expressStaticGzip('/static/', {
       enableBrotli: true,
     })
   );
 
+  let routes = [];
+
+  // Store all the registered express routes for usage with metrics reporting
+  routes = expressListEndpoints(app).map(endpoint => ({
```

Comment:
What is the value of `routes` after being set here? When I implemented this in the API, all the express routes had to be defined first before the `express-list-endpoints` lib could look them up. I'm not familiar with how routing works with the `next` js stuff.

Reply:
Good point, I copied this over and in this case `app` should be `server` (just noticed that). I'm not sure about the express routes with nextjs either. I'll look into it. @aulneau any thoughts? Should I do something like this: https://github.com/blockstack/stacks-blockchain-api/blob/22e9a4f64f093fd9982dde570e18278b65e4b934/src/api/init.ts#L113

Reply:
I am not too familiar with express, so was looking at documentation ...was going to do something like:

`server.use(
    (() => {
      const router = express.Router();
      router.use('/txid', next());
      router.use('/microblock', next());
      router.use('/block', next());
      router.use('/address', next());
      return router;
    })()
  );`